InnerEye/ML/deep_learning_config.py

```diff
@@ -135,8 +136,14 @@ def create(project_root: Path,
             else:
                 logging.info("All results will be written to a subfolder of the project root folder.")
                 root = project_root.absolute() / DEFAULT_AML_UPLOAD_DIR
-            timestamp = create_unique_timestamp_id()
-            run_folder = root / f"{timestamp}_{model_name}"
+            if is_global_rank_zero():
+                timestamp = create_unique_timestamp_id()
+                run_folder = root / f"{timestamp}_{model_name}"
+            else:
+                # Handle the case where there are multiple DDP threads on the same machine outside AML.
+                # Each child process will be started with the current working directory set to be the output
+                # folder of the rank 0 process. We want all other process to write to that same folder.
+                run_folder = Path.cwd().absolute()
```

Comment:
I probably miss some knowledge about what happens elsewhere - here it would seem the output of rank 0 and of other ranks will go to different outputs folders. Wouldn't we need to save the timestamp as environment variable to share across ranks? can you help me understanding this?

Reply:
Rank 0 sets the current working directory to be the timestamped folder. When the jobs with rank!=0 start, they see that timestamped folder as their current working directory - so they just have to keep on writing to cwd in order to write to exactly the same folder as rank 0.